### PR TITLE
Adding load_env.sh as an alternative to with-contenv

### PR DESCRIPTION
**Pull request by mayanand**
Purpose: To give an alternative to `with-contenv` tool for enviornment variables greater than 4K
* `with-contenv` tool, which is used to expose environment variables across scripts, has a limitation that it cannot read beyond 4k characters for environment variable values. To work around this issue, use the script `/scripts/load_env.sh` instead of `with-contenv`. You'll need to remove the `with-contenv` from the shebang line, and add  `source /scripts/load_env.sh` in the next line after the shebang line.

### CODE REVIEW

**Comment by mayanand** [comment]
@bryanlatten : Please take a look at this PR.

**Comment by bryanlatten** [comment]
👍

**Comment by bryanlatten** [comment]
@ko-be @bossjones @mannytoledo ?

**Comment by bossjones** [comment]
Can we add a couple goss test cases here?

1. To validate 4096 char works correctly
2. check to verify if `[ "${env_variable_name}" != "UID" ]` works.
3. check to verify `[ "${env_variable_name}" != "!" ]` works.
4. check to verify `[ "${env_variable_name}" != "_" ];` works.

**Comment by mayanand** [comment]
@bossjones this script `load_env.sh` will be executed during runtime and goss test is only used to test the docker build. can you provide an example of how we can incorporate these tests as per your suggestion?

**Comment by bryanlatten** [comment]
OoooOOOo: could be a good example to give https://github.com/aelsabbahy/goss/tree/master/extras/dgoss a shot, within the travis file

**Comment by mayanand** [comment]
@bryanlatten This looks interesting. Let me take a stab at it.

**Comment by mayanand** [comment]
@bryanlatten I have added the test case to check the truncation of env variable during runtime, within the travis file. But, it seems that dgoss is not installed on the system where travis job runs: https://travis-ci.org/behance/docker-base/builds/237697014#L771
This needs to installed using this: https://github.com/aelsabbahy/goss/tree/master/extras/dgoss#installation

@bossjones 
- To validate 4096 char works correctly : Added one test case each for success and failure.
- check to verify if [ "${env_variable_name}" != "UID" ] works. : This will always work because bash gives a syntax error when we try to export a variable name with `UID`. For instance
```
mayanand-macOS:docker-base mayanand$ export UID=22
-bash: UID: readonly variable
```
- check to verify [ "${env_variable_name}" != "!" ] works. : This will always work because bash gives a syntax error when we try export a variable name with `!`. For instance
```
mayanand-macOS:docker-base mayanand$ export !=0
-bash: export: `!=0': not a valid identifier
```
- check to verify [ "${env_variable_name}" != "_" ]; works. : Removed this condition as this is a valid operation.

**Comment by bryanlatten** [comment]
@mayanand 
1. you are able to pull and run `dgoss` within the travis environment.
2. it appears your baking those test scripts into the container. is it possible to keep them out, but mount them in during test-time? 
3. in yaml, you can possibly reuse a node, instead of copy-pasting a 4097 character env var: https://stackoverflow.com/questions/6651275/what-do-the-mean-in-this-database-yml-file

**Comment by mayanand** [comment]
@bryanlatten 
1. I didn't realize that earlier. I have done it now.
2. I have de-coupled the test scripts from the container and now I mount them during test-time.
3. I have reused a node but I still need to use the different o/p string in the test cases because they are not exactly same. I was looking for concatenation facility in YAML but it is not available by default. Link: https://stackoverflow.com/questions/40631913/concatenation-of-yaml-anchors-with-another-string-in-keys 
Please let me know if this looks good.

**Comment by bryanlatten** [comment]
Can I make one additional add? Can we rename the script to `with-bigcontenv` or something similar to the other script.

**Comment by mayanand** [comment]
@bryanlatten updated the file name and changed its file permissions.

**Comment by bryanlatten** [comment]
👍

**Comment by bryanlatten** [comment]
@bossjones can you take another look?

**Comment by bossjones** [comment]
Changes look excellent!!! @mayanand @bryanlatten last feedback, which should be quick. Address all the missing newline messages. I'll mark them specifically in the PR. When that's done this change is ready to get merged.

File: container/root/scripts/with-bigcontenv [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+#-------------------------------------------------------------------
+
+CONTAINER_ENV_LOC=/var/run/s6/container_environment/*
+
+for f in $CONTAINER_ENV_LOC; do
+  env_variable_name="${f##*/}"
+  if [ "${env_variable_name}" != "UID" ] && [ "${env_variable_name}" != "!" ]; then
+    export "${env_variable_name}"="`cat $f`"
+  fi
+done
```
Comment (bossjones) [inline container/root/scripts/with-bigcontenv]: Add new line here.

File: goss.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
...varlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargevarlargeva_extralargevar
```
Comment (bossjones) [inline goss.yaml]: Add new line here.

File: test.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+for x in Dockerfile*;do
+  echo -e "\n\n======================="
+  echo "testing $x"
+  echo "======================="
+  docker build -t runtimetestimage -f "${x}" .
+  #passes an env variable of size more than 4K chars
+  time dgoss run -v `pwd`/test_scripts:/test_scripts -e TEST_LONG_VAR="${largevar}" runtimetestimage || ((i++))
+done
+
+exit $i
```
Comment (bossjones) [inline test.sh]: Add new line here.

File: test_scripts/test_long_var_with_contenv.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/with-contenv bash
+
+echo ${TEST_LONG_VAR}
```
Comment (bossjones) [inline test_scripts/test_long_var_with_contenv.sh]: Add new line here.

File: test_scripts/test_long_var_without_contenv.sh [hunk 5]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash -e
+
+source /scripts/with-bigcontenv
+
+echo ${TEST_LONG_VAR}
```
Comment (bossjones) [inline test_scripts/test_long_var_without_contenv.sh]: Add new line here.

**Comment by mayanand** [comment]
@bossjones Added new line in the 5 files that you mentioned.

**Comment by bossjones** [comment]
perfect!

**Comment by bossjones** [comment]
👍

**Pull request merged**